abcix/adapter/adapter_test.go

```diff
@@ -0,0 +1,59 @@
+package adapter
+
+import (
+	"testing"
+
+	"github.com/magiconair/properties/assert"
+	abci "github.com/tendermint/tendermint/abci/types"
+	abcix "github.com/tendermint/tendermint/abcix/types"
+)
+
+type mockAbciApp struct {
+	abci.BaseApplication
+}
+
+func (app *mockAbciApp) CheckTx(req abci.RequestCheckTx) abci.ResponseCheckTx {
+	return abci.ResponseCheckTx{
+		Code: abci.CodeTypeOK,
+		Data: []byte("42"),
+	}
+}
+
+func (app *mockAbciApp) DeliverTx(req abci.RequestDeliverTx) abci.ResponseDeliverTx {
+	return abci.ResponseDeliverTx{
+		Data: req.Tx,
+	}
+}
+
+func (app *mockAbciApp) BeginBlock(req abci.RequestBeginBlock) abci.ResponseBeginBlock {
+	var events = make([]abci.Event, 1)
+	events[0] = abci.Event{Type: "begin"}
+	return abci.ResponseBeginBlock{Events: events}
+}
+
+func (app *mockAbciApp) EndBlock(req abci.RequestEndBlock) abci.ResponseEndBlock {
+	var events = make([]abci.Event, 1)
+	events[0] = abci.Event{Type: "end"}
+	return abci.ResponseEndBlock{Events: events}
```

Comment:
can also be simplified `return abci.ResponseEndBlock{Events: []abci.Event{{Type: "end"}}}`